### ISSUE TITLE
New TRV Thermostat Card causes high cpu load when status is "off"

### ISSUE THREAD

**Issue by phil11c** [home-assistant/frontend#18999]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Since 2023.12 my dashboards produce a high cpu load when TRVs are in status "off".
In the moment when automations change status to "heat", cpu load goes to 0.x %.

Tried with Chromium on Windows and HA-App on iOS.
Caches have been cleared.

Issue is (at least in my environment) reproducable.

### Describe the behavior you expected

Low cpu load regardless of TRVs status.

### Steps to reproduce the issue

1. Open dashboards with thermostat cards, wait for automation to turn of TRVs.
2. Watch task-manager or feel iPhone getting hot :-)
3. Wait until automation turn on TRVs again
4. Watch or feel the difference

### What version of Home Assistant Core has the issue?

2023.12.1

### What was the last working version of Home Assistant Core?

<2023.12

### In which browser are you experiencing the issue with?

Chromium 119.0.6045.110 (Official Build, ungoogled-chromium) (64-Bit) / Current HA-App

### Which operating system are you using to run this browser?

Windows 11 / iOS

**Comment by piitaya** [comment]
I can not reproduce. Can you restart Home Assistant in safe mode and try again? 

![CleanShot 2023-12-11 at 14 21 01](https://github.com/home-assistant/frontend/assets/5878303/5641d88b-829d-4059-a9a8-2395932c2c20)

**Comment by alflexRH** [comment]
I upgraded to 2023.12.1 again, and even in safe mode, i can reproduce it. A dashboard with no thermostat (but plenty of power related sensors) is fine. CPU load is low, dashboard is responsive and no errors scrolling by in the javascript console.
A dashboard with thermostat items gives a very high CPU load, is virtually unresponsive and the javascript console keeps scrolling these:
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	performUpdate (app.otA1ToECPHw.js:2:142696)
	(anonymous function) (app.otA1ToECPHw.js:2:142209)
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	(anonymous function) (app.otA1ToECPHw.js:2:142149)
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	performUpdate (app.otA1ToECPHw.js:2:142696)
	(anonymous function) (app.otA1ToECPHw.js:2:142209)
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	(anonymous function) (app.otA1ToECPHw.js:2:142149)
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	performUpdate (app.otA1ToECPHw.js:2:142696)
	(anonymous function) (app.otA1ToECPHw.js:2:142209)
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	(anonymous function) (app.otA1ToECPHw.js:2:142149)
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	performUpdate (app.otA1ToECPHw.js:2:142696)
	(anonymous function) (app.otA1ToECPHw.js:2:142209)
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	(anonymous function) (app.otA1ToECPHw.js:2:142149)
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	performUpdate (app.otA1ToECPHw.js:2:142696)
	(anonymous function) (app.otA1ToECPHw.js:2:142209)
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	(anonymous function) (app.otA1ToECPHw.js:2:142149)
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	performUpdate (app.otA1ToECPHw.js:2:142696)
	(anonymous function) (app.otA1ToECPHw.js:2:142209)
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	(anonymous function) (app.otA1ToECPHw.js:2:142149)
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	performUpdate (app.otA1ToECPHw.js:2:142696)
	(anonymous function) (app.otA1ToECPHw.js:2:142209)
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	(anonymous function) (app.otA1ToECPHw.js:2:142149)
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	performUpdate (app.otA1ToECPHw.js:2:142696)
	(anonymous function) (app.otA1ToECPHw.js:2:142209)
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	(anonymous function) (app.otA1ToECPHw.js:2:142149)
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	performUpdate (app.otA1ToECPHw.js:2:142696)
	(anonymous function) (app.otA1ToECPHw.js:2:142209)
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	(anonymous function) (app.otA1ToECPHw.js:2:142149)
[Error] Unhandled Promise Rejection: SyntaxError: The string did not match the expected pattern.
	performUpdate (app.otA1ToECPHw.js:2:142696)
	(anonymous function) (app.otA1ToECPHw.js:2:142209)

**Comment by phil11c** [comment]
Thanks alflexRH,
i can confirm that safe mode shows the same problem.

**Comment by karwosts** [comment]
Maybe someone grab the state/attributes for the climate entity that causes this from developer-tools?

**Comment by alflexRH** [comment]
I have narrowed it down to 2 heating devices, 1 that does cause the issue, another that doesnt. 
The one that does is a Zigbee Z-TRM3 by Heatit (called floor_thermostat). A LG aircon via the "SmartThinQ LGE Sensors" HACS module does not cause the problem (called bedroomaircon). So i now have 2 dashboards with just 1 sensor each as an example.

in the Developer Tools -> States menu, i dont see a neat way to copy&paste the states however, but when zooming out a bit, it just fits in 1 screen.

I hope this helps. 

<img width="959" alt="screenshot_bedroomaircon" src="https://github.com/home-assistant/frontend/assets/42237394/6e9adedd-814d-4ad5-a586-1be3cebc9c49">
<img width="935" alt="screenshot_floor_thermostat" src="https://github.com/home-assistant/frontend/assets/42237394/bfac962e-ac18-4a6a-a5ce-068159fcfbef">

**Comment by alflexRH** [comment]
And with this test setup i can also confirm that the high cpu load (and javascript console errors) go away when i turn the heating on. 

The temperature attribute going from NULL to its setpoint, is the only obvious change i see.

<img width="902" alt="Screenshot 2023-12-12 at 21 27 17" src="https://github.com/home-assistant/frontend/assets/42237394/7cdfe16e-6d9f-4ac7-9327-f552341b76d9">

**Comment by karwosts** [comment]
I don't see high cpu usage, but I do see a loop of quite a lot of errors when I set a climate attribute to `temperature: null`, that could be possibly related. 

If you take your off climate entity, click the blue link there, and overrride `temperature: null` to some valid number, does the high cpu usage go away? (You can temporarily override attributes in this developer tools page).

**Comment by alflexRH** [comment]
yup, thats it. i have the dashboard and state editor open in 2 separate tabs. when its null, the errors scroll by and the cpu load is high, when i chance it to a number, the messages stop and the cpu load is normal. in addition, when i change it back to null, everything remains ok.

**Comment by Ferrox85** [comment]
Please do a quick fix on this. It makes my home assistant front end freeze!
I added 3 thermostats in a horizontal stack. If I delete this from the dashboard my home assistant instantly becomes responsive again.

**Comment by drsgoodall** [comment]
I am seeing similar issues but with the javascript message "Uncaught (in promise) DOMException: Failed to execute 'remove' on 'DOMTokenList': The token provided must not be empty."

Checking "states" in developer tools I can also see the temperature becomes null when the heating mode is off. 

I can also see the same errors occurring on the demo page - https://design.home-assistant.io/#more-info/climate 

On the default dashboard I have about 15 controls and these error messages appear in a constant stream - hundreds in a matter of seconds.

**Comment by developerfromjokela** [comment]
Same issue here with zigbee TS0601 thermostats. UI becomes very sluggish, unusuable.

**Comment by developerfromjokela** [comment]
> I am seeing similar issues but with the javascript message "Uncaught (in promise) DOMException: Failed to execute 'remove' on 'DOMTokenList': The token provided must not be empty."
> 
> Checking "states" in developer tools I can also see the temperature becomes null when the heating mode is off.
> 
> I can also see the same errors occurring on the demo page - https://design.home-assistant.io/#more-info/climate
> 
> On the default dashboard I have about 15 controls and these error messages appear in a constant stream - hundreds in a matter of seconds.

Just go there and turn off, same errors appear. Is somebody doing a PR or fix atm?

**Comment by marc4s** [comment]
absolutly worst case on android phone with a dashboard using 4 such new Thermostat Cards. Thanks to this issue I could remove them. I was searching for the root cause already a while. But from my point of view it was fine till 2023.2 and I got affacted with 2023.3(or maybe affected after a change to previous hidden functions which I activated. But sorry I can't remember anymore if this behaviour came up directly after adding the functions or not...I think not)
